### ISSUE TITLE
Dust3r Unable to Deal with Many Images

### ISSUE THREAD

**Issue by eyildiz-ugoe** [naver/dust3r#111]
I'm using the provided demo script to feed in around 300 images (with 2K resolution, though they get reduced to 512x512 anyway) to get a precise 3D model of the scene. The inference takes forever, I've launched it like 1h ago and it says it is still at 30% on inference. Apparently there are 107256 image pairs (?) and it is unable to process all. I tried it a few times, **it always comes to 30% and then it gets stuck**, the screen is not interactive anymore.  My computer is not even bad, I have 3090RTX and a lot of RAM and a decent CPU. So it's not about the hardware at all.

Either the demo parameters are off, or the implementation has a terrible bottleneck the authors are not aware of. 

In any case, this is a problem. Just letting you know so that you refactor it on your side.

**Comment by arminak6** [comment]
Yes, it is a problem, and you cannot work with many images. They are working on an optimized solution for this issue. By the way, for `scene_graph,` you can use 'swin' or 'oneref' to find a smaller number of pairs, resulting in less computation.
